Tests/StreamChatTests/Controllers/ChannelWatcherListController/ChatChannelWatcherListController_Tests.swift

```diff
@@ -266,6 +266,8 @@ final class ChatChannelWatcherListController_Tests: XCTestCase {
 
         // Simulate network call response
         env.watcherListUpdater!.channelWatchers_completion!(nil)
+        // State change is only notified when observer is called with data changes.
+        controller.watchersObserver.onDidChange?([])
```

Comment:
Not sure if we should expose the observers to the tests. If you want to simulate sth, maybe it's better to send an event via the event notification center? Same applies for the other tests.

Reply:
It is fine to expose the observers internally, actually the tests should be like this in the first place, but they would need to be injected in the initialiser, which at the moment is not possible. Either way, since were going to do this only in the BG Mapping, I will remove these tests